python/rmm/_lib/lib.pyx

```diff
@@ -0,0 +1,325 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from rmm._lib.lib cimport *
+from libc.stdint cimport uintptr_t
+from libc.stdlib cimport malloc, free
+from libcpp.vector cimport vector
+
+
+# Global options, set on initialization, freed on finalization
+cdef rmmOptions_t *opts = NULL
+
+
+# Utility Functions
+def _get_error_msg(errcode):
+    """
+    Get error message for the given error code.
+    """
+    if errcode == RMM_ERROR_CUDA_ERROR:
+        # Need to move code from libcudf into librmm for this?
+        # See cpp/src/utilities/cuda_utils.cu
+        # cudaerr = self._api.gdf_cuda_last_error()
+        # errname = self._ffi_str(self._api.gdf_cuda_error_name(cudaerr))
+        # details = self._ffi_str(self._api.gdf_cuda_error_string(cudaerr))
```

Comment:
This abused CFFI for a circular dependency on libcudf, need to decide what we want to do here...

Reply:
@harrism could use your guidance here on what we should do for error checking the functions for now. It looks like we were checking for libgdf error codes within RMM here which feels... wrong?

Reply:
No, we were checking *CUDA* error codes using cudf's `gdf_cuda_error_*()` functions. All those do is pass through to CUDA APIs.  I think when we switch to exceptions the error codes go away in RMM, and it will throw exceptions with useful messages anyway. So this whole function can go away then, right?

Reply:
You are right that was a silly dependency on cuDF.

Reply:
Connected with @harrism on a call to discuss and the plan of action is to merge #38 into this PR, and then remove all of the error code checking since we'll be throwing exceptions. We can move from the C APIs to the C++ APIs in the future follow up.

Reply:
cc @jrhemstad if you have any thoughts on this since #38 has been sitting on deck for quite a bit.

Reply:
Connected with @jrhemstad and he said #38 isn't feasible for 0.10, so just moved to using the built-in RMM functions for handling RMM error codes and we can rip that out in the future when we enable exceptions.